engine/movie/opening_cutscene.asm

```diff
@@ -0,0 +1,1235 @@
+	ret
+
+.scene3_5:
+	xor a
+	ldh [hLCDCPointer], a
+	ret
+
+.scene3_6:
+	call Intro_UpdateLYOverrides
+	ret
+
+Intro_InitBubble:
+	ld hl, wIntroFrameCounter1
+	ld a, [hl]
+	and $0f
+	ret nz
+
+	ld a, [hl]
+	and $70
+	swap a
+	ld e, a
+	ld d, $00
+	ld hl, .pixel_table
+	add hl, de
+	add hl, de
+	ld e, [hl]
+	inc hl
+	ld d, [hl]
+	ld a, SPRITE_ANIM_INDEX_GS_INTRO_BUBBLE
+	call InitSpriteAnimStruct
+	ret
+
+.pixel_table
+	dbpixel  6, 14,  0,  4
+	dbpixel 14, 18,  0,  4
+	dbpixel 10, 16,  0,  4
+	dbpixel 12, 15,  0,  0
+	dbpixel  4, 13,  0,  0
+	dbpixel  8, 17,  0,  0
+
+Intro_InitMagikarps:
+	depixel 8, 7, 0, 7
+	ld a, [wSGB]
+	and a
+	jr z, .ok
+	depixel 4, 3, 0, 7
+
+.ok
+	ld hl, wIntroFrameCounter2
+	ld a, [hl]
+	and e
+	ret nz
+	ld a, [hl]
+	and d
+	jr nz, .alternate_frame
+
+	depixel 29, 28
+	call .PlaceMagikarp
+
+	depixel 26, 0
+	call .PlaceMagikarp
+
+	depixel 0, 24
+	call .PlaceMagikarp
+	ret
+
+.alternate_frame
+	depixel 28, 30
+	call .PlaceMagikarp
+
+	depixel 31, 24
+	call .PlaceMagikarp
+
+	depixel 2, 28
+	call .PlaceMagikarp
+	ret
+
+.PlaceMagikarp:
+	ld a, SPRITE_ANIM_INDEX_GS_INTRO_MAGIKARP
+	call InitSpriteAnimStruct
+	ret
+
+Intro_InitOmanyte:
+	depixel 18, 7
+	call .PlaceOmanyte
+	depixel 14, 10
+	call .PlaceOmanyte
+	depixel 16, 15
+; fallback
+
+.PlaceOmanyte:
+	ld a, SPRITE_ANIM_INDEX_GS_INTRO_OMANYTE
+	call InitSpriteAnimStruct
+	ret
+
+Intro_InitLapras:
+	ld a, [wIntroFrameCounter2]
+	and %00011111
+	ret nz
+	depixel 16, 24
+	ld a, SPRITE_ANIM_INDEX_GS_INTRO_LAPRAS
+	call InitSpriteAnimStruct
+	ret
+
+Intro_UnusedInitAerodactyl:	; unreferenced
+	depixel 2, 0
+	ld a, SPRITE_ANIM_INDEX_UNUSED_INTRO_AERODACTYL
+	call InitSpriteAnimStruct
+	ret
+
+Intro_UpdateTilemapAndBGMap:
+; add new tiles to top as water scene scrolls up to surface
+	push hl
+	push de
+
+	ld a, [wIntroTilemapPointer + 0]
+	ld e, a
+	ld a, [wIntroTilemapPointer + 1]
+	ld d, a
+	ld hl, -$10
+	add hl, de
+	ld a, l
+	ld e, l
+	ld [wIntroTilemapPointer + 0], a
+	ld a, h
+	ld d, h
+	ld [wIntroTilemapPointer + 1], a
+
+	hlcoord 0, 0
+	ld c, BG_MAP_WIDTH / 2
+.loop
+	call Intro_Draw2x2Tiles
+	dec c
+	jr nz, .loop
+
+	ld a, [wIntroBGMapPointer + 0]
+	ld e, a
+	ld a, [wIntroBGMapPointer + 1]
+	ld d, a
+	ld hl, hCurSpriteYCoord
+	add hl, de
+	ld a, l
+	ld [wIntroBGMapPointer + 0], a
+	ld [wVBCopyDst], a
+	ld a, h
+	and %11111011
+	or %00001000
+	ld [wIntroBGMapPointer + 1], a
+	ld [wVBCopyDst + 1], a
+	ld a, LOW(wTileMap)
+	ld [wVBCopySrc], a
+	ld a, HIGH(wTileMap)
+	ld [wVBCopySrc + 1], a
+	ld a, 4
+	ld [wVBCopySize], a
+	ld hl, wIntroFrameCounter1
+	dec [hl]
+	pop de
+	pop hl
+	ret
+
+
+Intro_AnimateOceanWaves:
+; uses a 2bpp request to copy tile IDs to the BG map
+	ld hl, wIntroFrameCounter2
+	ld a, [hl]
+	and 3
+	cp 3
+	ret z
+
+	ld a, [hl]
+	and $30
+	swap a
+	ld l, a
+	ld h, 0
+rept 5
+	add hl, hl
+endr
+	ld de, .wave_tiles
+	add hl, de
+	ld a, l
+	ld [wVBCopySrc], a
+	ld a, h
+	ld [wVBCopySrc + 1], a
+	ld a, LOW(vBGMap0 + (15 * BG_MAP_WIDTH)) ; vBGMap0 row 15
```

Comment:
"`<data> + <n> * <width>`" to get the `<n>`th row of `<data>` is a pretty common pattern; no need for the comments